### ISSUE TITLE
Database design

### ISSUE THREAD

**Issue by Muq2** [CSAllenISD/2023-ISP-ModelScheduler#2]
![image](https://user-images.githubusercontent.com/89667982/211946596-4b817b26-df4b-47a9-8efb-0da660fdd5f5.png)

What other data will we have/need in the courses and student tables?

**Comment by ryanhallock** [comment]
If we are starting with a MVP, we only require courses. I am assuming that you want to use a bit mask for size & period. 
Also the course ID's provided are in the `string` format if we plan on showing the user the id they need to search in arena.  We might also have issues with special cases classes (dual blocked, DC classes).

After MVP is created and we decide to do analytics and show users other students using the course we would need the other tables, but this might be problematic storing user data. The problem is creating a unique user, we do not want to store the email or student id. We maybe could store a unique hash of there email to associate them, and tossing the original email. We would also need seat size, and seats renaming to show them this feature.

TLDR: MVP only needs courses with a course id `string`, storing emails might be problematic, and showing students size requires seat size and seats renaming.

**Comment by Muq2** [comment]
Mr BY put forward the idea of having accounts. When the account selects the course they want the course size will increment, showing other users how many people are interested in the course.

**Comment by Muq2** [comment]
Also i think that we should have a separate incremental id for each course, making it more accessible later on. We can store the actual ahs class code as a string in the courses table.

**Comment by Muq2** [comment]
EX:
ID = 1
ClassCode = TA2S2A 
Name = Comp Sci 2

**Comment by Muq2** [comment]
Technically we dont need anything in the Student Courses table other than accountid and course_id. We can lookup the course data from the course table using the course_id when needed.

**Comment by Reynard-G** [comment]
Do we even need the primary key as a made-up ID for each class? Why not use the class code as the primary key to reduce the number of columns.

**Comment by Muq2** [comment]
We could, I just don't like dealing with strings as primary keys

**Comment by ryanhallock** [comment]
Strings as primary keys could work better along as we don't duplicate keys when deleting/adding new classes. Do we have a strategy if students select a class period that changes later on?

**Comment by MuqDonalds** [comment]
elaborate

**Comment by Muq2** [comment]
ok i will change the primary key to the class code string. what do you mean by if a period changes?

**Comment by Muq2** [comment]
i believe each period is it’s own course with its own class code

**Comment by Muq2** [comment]
Once we finalize the database design i can create the sql script

**Comment by ryanhallock** [comment]
While getting closer to Arena Scheduling, the database (at least last year) kept getting updates. I was wondering how we could handle such things. Most likely an issue that should be tested but will never happen in practice.

**Comment by Reynard-G** [comment]
You could try to run a script that validates the classes selected everytime the database is updated. If it's not valid, remove the class from the student's selected class.

**Comment by Muq2** [comment]
Yes we can do that on load

**Comment by JacobJershin** [comment]
I like it 😄 👍

**Comment by Muq2** [comment]
If you guys have anything else you think we might need please tell me

**Comment by Boden-C** [comment]
for mvp, maybe add location and credits? those are required to check for conflicts and a valid schedule
if were doing a bitmap for period, should prob rename it to periodBitmap

that should be the very minimum for arena scheduling. the goal of this model scheduler is to be more reactive though, so were probably gonna have to add more fields once we figure out what the main goals are (i floated around some ideas but they won't work until he talks with admin)

**Comment by Muq2** [comment]
![image](https://user-images.githubusercontent.com/89667982/211845291-41c596c2-436c-494f-b302-8a606c925725.png)

**Comment by sap786** [comment]
keep in mind we cant store student personal details (ex; email or student id)

**Comment by Muq2** [comment]
> keep in mind we cant store student personal details (ex; email or student id)

that is a placeholder until BY figures out how to bypass feels

**Comment by JacobJershin** [comment]
Do we need emails? Could we just used username and password? it's not like we'd need their email for anything other then if they forgot their user name or password and need a verification code or whatever but do we really need that? People using the mock scheduler are gonna use it like 2 or 3 times and are most likely just gonna use their school ID and password so the chances of them needing a verification email to be sent are very low. I totally get using it though and I'm not against using it I'm just wondering if we should.
